### ISSUE TITLE
Beam decoding?

### ISSUE THREAD

**Issue by suryabhupa** [spro/torch-seq2seq-attention-translate#1]
I haven't dug into the code too much, but is there support for beam decoding?

**Comment by spro** [comment]
There isn't, but I would highly recommend the yet-unpublished drop-output technique for sequential decoders, where you put dropout [right before the softmax](https://github.com/spro/torch-seq2seq-attention/blob/406e79b2920726aa29de132150fe9d0abbaf2ff6/train.lua#L35) to get a similar result with less computation.

**Comment by suryabhupa** [comment]
Oh dang, that's pretty sneaky (and great). Do you happen to have numbers on how this compares to normal beam search?

**Comment by spro** [comment]
No numbers yet, I really just stumbled upon the idea by misusing dropout once. Anecdotally it worked great for increasing the variety of dialog responses.
